### PR TITLE
Add simd_op_check tests and a few more patterns

### PR DESCRIPTION
**Pull request by dsharletg**
We care about count_leading_zeros, and this PR fills in some other gaps in coverage. Also add a few patterns, I think we should have non-rounding versions of instructions we have rounding patterns for.

### CODE REVIEW

File: test/correctness/simd_op_check.cpp [hunk 1]
```diff
@@ -1909,6 +1937,10 @@ int main(int argc, char **argv) {
         test.set_num_threads(1);
     }
 
+    if (getenv("FILTER")) {
```
Comment (steven-johnson) [inline test/correctness/simd_op_check.cpp:1940]: nit: is this really an improvement over the optional command-line arg we already have?

File: src/CodeGen_ARM.cpp [hunk 2]
```diff
@@ -149,6 +149,18 @@ CodeGen_ARM::CodeGen_ARM(Target target)
         }
     }
 
+    casts.emplace_back("vqdmulh.v4i16", "sqdmulh.v4i16", 4,
```
Comment (steven-johnson) [inline src/CodeGen_ARM.cpp:152]: So this PR is really "add some cast cases we didn't handle before, plus testing for them in simd_op_check"? Probably worth updating the commit/PR title then, I had assumed the change was just to the test, not the codegen.

**Review by steven-johnson — APPROVED** [review]
LGTM with nits.

File: test/correctness/simd_op_check.cpp (hunk 1 above)
Comment (dsharletg) [reply test/correctness/simd_op_check.cpp:1940]: Err, I forgot to remove this. But yes, it is, I add it locally every time I'm doing something with simd_op_check. You can't pass a command line arg to the makefile. It's a few extra steps to work around, and running simd_op_check drops a lot of files wherever you run it from.

Comment (abadams) [reply test/correctness/simd_op_check.cpp]: Perhaps simd_op_check shouldn't do that. I've had it bomb my working directory in that way too, and it's super annoying. It should just make a tmp subdir (and print what it is so you can go inspect the files if you really want to)

Comment (steven-johnson) [reply test/correctness/simd_op_check.cpp]: +100, output to /tmp dir always would be much better.

re: "FILTER", can we at least make it something slightly less generically named?

**Comment by steven-johnson** [comment]
simd_op_check is failing on arm32: https://buildbot.halide-lang.org/master/#/builders/38/builds/10

**Comment by dsharletg** [comment]
This PR only affects ARM, failures must be unrelated.

**Pull request merged**